src/tpm2tss-genkey.c

```diff
@@ -207,6 +221,12 @@ parse_opts(int argc, char **argv)
         ERR("%s", help);
         exit(1);
     }
+
+    if ((opt.importpub && !opt.importtpm) || (!opt.importpub && opt.importtpm)) {
+        ERR("--public requires --private");
```

Comment:
The message depends on which one is missing, so you can get the wrong message here.

```C
/* XXX I think this will do it */
if (opt.importpub != opt.importpub) {
   ERR (opt.importpub ? "--public requires --private" :
       "--private requires --public");
}

Reply:
The opt.importpub is a char pointer-- the (opt.importpub != opt.importpub) will be always true, unless both are NULL. We could either have two conditions, or change the error message to say, e.g.:
```C
    if ((opt.importpub && !opt.importtpm) || (!opt.importpub && opt.importtpm)) {
        ERR("Import requires both --public and --private");
        return 1;
    }
```

Reply:
I just pushed another revision that somehow combines both approaches.

Reply:
> I just pushed another revision that somehow combines both approaches.

glad you sorted it out, yeah I forgot to convert them to boolean values, but I see you nailed it with the ``!!``.